src/hooks/use-dnd-sensor.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useCallback, useMemo } from 'react'
+import {
+  useSensor,
+  useSensors,
+  PointerSensor,
+  TouchSensor,
+  DragStartEvent,
+  DragEndEvent,
+  Active
+} from '@dnd-kit/core'
+import { arrayMove } from '@dnd-kit/sortable'
+
+type IdProp = 'id' | '_id'
+
+const useDndSensor = <T extends { [K in IdProp]?: string }>(
+  items: T[],
+  setItems: (items: T[]) => void,
+  idProp: IdProp
```

Comment:
Maybe it would be better interface for hook if we use object instead of three different parameters? I mean usage like following:

```jsx
const {...} = useDndSensor({
    items: ...,
    setItems: ...,
    idProp: ...
})
```

What do you think about it?

Reply:
Thank you, it's cool idea! Refactored